src/spootnik/reporter/impl.clj

```diff
@@ -85,13 +102,13 @@
   [_]
   (throw (ex-info "Cannot build riemann client for invalid protocol" {})))
 
-(defn- build-metrics-reporter-dispatch-fn [_ _ _ [k _]]
+(defn- build-metrics-reporter-dispatch-fn [_ _ _ _ [k _]]
   (info "building" k "reporter!")
   k)
 
 (defmulti build-metrics-reporter build-metrics-reporter-dispatch-fn)
 
-(defmethod build-metrics-reporter :console [reg _ _ [_ {:keys [opts interval]}]]
+(defmethod build-metrics-reporter :console [reg _ _ _ [_ {:keys [opts interval]}]]
```

Comment:
> Switching this to a cond ties us into supporting all possible reporter types from the main module, is this what we want?

Actually, this was my assumption. The main module is responsible for handling all the different reporter types.
For me, this represents the current situation and we used to use a `defmulti` because it was more elegant. In practice, it doesn't seem to be the case, because each time we had a new reporter, we change the signature of the `defmulti`.
It happened with `Riemann` support, then with `Prometheus` and today with `Push Gateway`.

Otherwise, if we want to keep the `defmulti`, I would then revisit the signature to have something stable, like introducing `deps` which can be derestructured to get the dependencies out of it... (but yeah... not crazy)

```clojure
(defn- build-metrics-reporter-dispatch-fn [reg deps [type opt]]
  (info "building" type "reporter!")
  type)
``` 

> It's up to the "edge" to then decide where and how to report the metrics. Which is what the defmulti is here to cover: based on the provided configuration, it sets up reporting to the correct place.

I think I get it, but not sure it was even used that way.
- `defmulti` is not stable at the moment, so the edge would break each time we add a new reporter
- If I want to add a new reporter of `:new-reporter`, I'm pretty sure that I will need again another dependency that should be provided by the main module.

Or I may miss something....